edx_event_bus_kafka/internal/config.py

```diff
@@ -38,12 +38,27 @@ def get_schema_registry_client():
         warnings.warn('Library confluent-kafka not available. Cannot create schema registry client.')
         return None
 
+    # .. setting_name: EVENT_BUS_KAFKA_SCHEMA_REGISTRY_URL
+    # .. setting_default: None
+    # .. setting_description: URL of the Avro schema registry, required for managing the evolution
+    #   of event bus data and ensuring that consumers are able to decode the events that
+    #   are produced. This URL is required for both producers and consumers and must point
+    #   to an instance of Confluent Schema Registry:
+    #   https://docs.confluent.io/platform/current/schema-registry/index.html
     url = getattr(settings, 'EVENT_BUS_KAFKA_SCHEMA_REGISTRY_URL', None)
     if url is None:
         warnings.warn("Cannot configure event-bus-kafka: Missing setting EVENT_BUS_KAFKA_SCHEMA_REGISTRY_URL")
         return None
 
+    # .. setting_name: EVENT_BUS_KAFKA_SCHEMA_REGISTRY_API_KEY
+    # .. setting_default: None
+    # .. setting_description: API key for talking to the Avro schema registry specified in
```

Comment:
This key and secret are optional as well (figure we should probably be consistent with how we describe the Kafka API key)

Reply:
Oh! Yes, I was wondering about this and forgot to leave myself a note. The Kafka API key/secret are definitely optional because if either one is omitted, we skip adding both of them to the config. But with these schema registry values, we _always_ use them, even if they're None or empty. So for instance we could end up with a `basic.auth.user.info` of `:`, `user:`, `:pass`, or `user:pass`. Not all of those look particularly sensible. I'll go check the schema registry docs again...

(Also, I see that I've documented them as defaulting to `None` but they're actually `''`. Oops.)

Reply:
This checks if user info = ('','') (ie we passed ':') so I think we're fine https://github.com/confluentinc/confluent-kafka-python/blob/bbe1225dc7d36df2c7e3e0f7476bb283fabcd1ff/src/confluent_kafka/schema_registry/schema_registry_client.py#L101

Reply:
It looks like it will look to the url first for auth information and then error if there is also `basic.auth.user.info.` I vote we punt on what happens if the schema registry server doesn't expect any auth and just put in docs somewhere that user should use the settings provided instead of putting the user and password in the URL

Reply:
Yeah, we should probably document that auth goes in the auth settings and not in the URL. But what to do about required/optional? I don't want to make claims in the docstrings that we can't back up, and so I'm tempted to just... not mention required/optional either way, for those two.

Reply:
I think "Optional" still makes sense because we do know for sure that you don't have to always have to set them.

Reply:
Hmm. I still don't know what happens if the schema registry isn't configured for auth and we pass an auth header containing `':'` anyway. Maybe nothing? I guess if it causes any issues then we can put in a bugfix when we find out.